corehq/motech/repeaters/dbaccessors.py

```diff
@@ -88,13 +94,13 @@ def _get_startkey_endkey_all_records(domain, repeater_id=None, state=None, dates
         kwargs['startkey'] = [domain, repeater_id, state, startdatekey]
     elif state:
         kwargs['endkey'] = [domain, repeater_id, state]
-        kwargs['startkey'] = [domain, repeater_id, state, None]
+        kwargs['startkey'] = [domain, repeater_id, state, {}]
     elif repeater_id:
         kwargs['endkey'] = [domain, repeater_id]
-        kwargs['startkey'] = [domain, repeater_id, None, None]
+        kwargs['startkey'] = [domain, repeater_id, None, {}]
     else:
         kwargs['endkey'] = [domain, None, None]
-        kwargs['startkey'] = [domain, None, None, None]
+        kwargs['startkey'] = [domain, None, None, {}]
```

Comment:
For the startkey the queries work with null and {}; but if I use {} for endkey (happens [here](https://github.com/dimagi/commcare-hq/pull/32626/files#diff-3a8c7e32fc20d66cf08a263be8665458fc24303fb74500e26a1e2ae35cb9b1f8R102) when only domain is provided), it gives a query error. So I kept this as it is.

For e.g.

```
======================================================================
ERROR: corehq.motech.repeaters.tests.test_dbaccessors:TestRepeatRecordDBAccessors.test_get_paged_repeat_records_wrong_domain
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sravan/HQ/commcare-hq/corehq/motech/repeaters/tests/test_dbaccessors.py", line 135, in test_get_paged_repeat_records_wrong_domain
    records = get_paged_repeat_records('wrong-domain', 0, 2)
  File "/Users/sravan/HQ/commcare-hq/corehq/motech/repeaters/dbaccessors.py", line 115, in get_paged_repeat_records
    return get_paged_couch_repeat_records(domain, skip, limit, repeater_id, state,
  File "/Users/sravan/HQ/commcare-hq/corehq/motech/repeaters/dbaccessors.py", line 131, in get_paged_couch_repeat_records
    results = RepeatRecord.get_db().view('repeaters/repeat_records_by_last_checked', **kwargs).all()
  File "/Users/sravan/.pyenv/versions/cchq/lib/python3.9/site-packages/couchdbkit/client.py", line 940, in all
    return list(self.iterator())
  File "/Users/sravan/.pyenv/versions/cchq/lib/python3.9/site-packages/couchdbkit/client.py", line 899, in iterator
    self._fetch_if_needed()
  File "/Users/sravan/.pyenv/versions/cchq/lib/python3.9/site-packages/couchdbkit/client.py", line 978, in _fetch_if_needed
    self.fetch()
  File "/Users/sravan/.pyenv/versions/cchq/lib/python3.9/site-packages/couchdbkit/client.py", line 957, in fetch
    self._result_cache = self.fetch_raw()
  File "/Users/sravan/.pyenv/versions/cchq/lib/python3.9/site-packages/couchdbkit/client.py", line 974, in fetch_raw
    return self._fetch(self._arg, self.params)
  File "/Users/sravan/.pyenv/versions/cchq/lib/python3.9/site-packages/couchdbkit/client.py", line 758, in raw_view
    return view(**params)
  File "/Users/sravan/.pyenv/versions/cchq/lib/python3.9/site-packages/cloudant/view.py", line 236, in __call__
    resp = get_docs(self._r_session,
  File "/Users/sravan/.pyenv/versions/cchq/lib/python3.9/site-packages/cloudant/_common_util.py", line 262, in get_docs
    resp.raise_for_status()
  File "/Users/sravan/.pyenv/versions/cchq/lib/python3.9/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request query_parse_error No rows can match your key range, reverse your start_key and end_key or set descending=false for url: http://localhost:5984/test_commcarehq__receiverwrapper/_design/repeaters/_view/repeat_records_by_last_checked?include_docs=true&reduce=false&limit=2&skip=0&descending=true&endkey=%5B%22wrong-domain%22%2C+null%2C+%7B%7D%5D&startkey=%5B%22wrong-domain%22%2C+null%2C+null%2C+null%5D
```

Reply:
Ah, I see now that the query is always using descending sort order. This makes the `_get_startkey_endkey_all_records` quite confusing. Can you add that do the docstring?

Ideally I think it would be better to assume normal sort order in this function and then the caller can decide what to do but that's a larger change.

Reply:
I have added the docstring.